datafusion/core/src/physical_optimizer/aggregate_statistics.rs

```diff
@@ -273,6 +263,44 @@ fn take_optimizable_max(
     None
 }
 
+fn is_non_distinct_count(agg_expr: &dyn AggregateExpr) -> bool {
```

Comment:
I really like the encapsulation of this logic into functions 👍 

```suggestion
// TODO: Move this check into AggregateUDFImpl
// https://github.com/apache/datafusion/issues/11153
fn is_non_distinct_count(agg_expr: &dyn AggregateExpr) -> bool {
```